specs/CookieManagement.md

```diff
@@ -0,0 +1,505 @@
+{
+    //! [GetCookies]
+    if (m_cookieManager)
+    {
+        CHECK_FAILURE(m_cookieManager->GetCookies(
+            uri.c_str(),
+            Callback<ICoreWebView2GetCookiesCompletedHandler>(
+                [this, uri](HRESULT error_code, ICoreWebView2CookieList* list) -> HRESULT {
+                    CHECK_FAILURE(error_code);
+
+                    std::wstring result;
+                    UINT cookie_list_size;
+                    CHECK_FAILURE(list->get_Count(&cookie_list_size));
+
+                    if (cookie_list_size == 0)
+                    {
+                        result += L"No cookies found.";
+                    }
+                    else
+                    {
+                        result += std::to_wstring(cookie_list_size) + L" cookie(s) found on " +
+                                  uri + L".";
+                        result += L"\n\n[";
+                        for (int i = 0; i < cookie_list_size; ++i)
+                        {
+                            wil::com_ptr<ICoreWebView2Cookie> cookie;
+                            CHECK_FAILURE(list->GetValueAtIndex(i, &cookie));
+
+                            if (cookie.get())
+                            {
+                                result += CookieToJsonString(cookie.get());
+                                if (i != cookie_list_size - 1)
+                                {
+                                    result += L",\n";
+                                }
+                            }
+                        }
+                        result += L"]";
+                    }
+                    MessageBox(nullptr, result.c_str(), L"GetCookies Result", MB_OK);
+                    return S_OK;
+                })
+                .Get()));
+    }
+    //! [GetCookies]
+}
+```
+
+## .NET and WinRT
+
+```c#
+void AddOrUpdateCookieCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    CoreWebView2Cookie cookie = webView.CoreWebView2.CookieManager.CreateCookie("CookieName", "CookieValue", ".bing.com", "/");
+    cookie.SameSite = CoreWebView2CookieSameSiteKind.None;
+    webView.CoreWebView2.CookieManager.AddOrUpdateCookie(cookie);
+}
+
+async void GetCookiesCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    IList<CoreWebView2Cookie> cookieList = await webView.CoreWebView2.CookieManager.GetCookiesAsync("https://www.bing.com");
+    for (uint i = 0; i < cookieList.Count; ++i)
+    {
+        CoreWebView2Cookie cookie = cookieList[i];
+        Cookie systemNetCookie = cookie.ToSystemNetCookie();
+        Console.WriteLine(systemNetCookie.ToString());
+    }
+}
+
+void DeleteAllCookiesCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    webView.CoreWebView2.CookieManager.DeleteAllCookies();
+}
+
+void DeleteCookiesCmdExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    webView.CoreWebView2.CookieManager.DeleteCookies("CookieName", "https://www.bing.com");

```

Comment:
It's weird that you get and delete with a URI (including scheme), but you add with a domain and path (no scheme).

How does the scheme interact with the IsSecure property? Suppose I have two cookies associated with `bing.com`, one secure and one not. Which ones get deleted when I delete with `https`? Which ones get deleted when I delete with `http`?

| URI passed to DeleteCookies | Secure cookie | Insecure cookie |
|-|-|-|
| `https://bing.com/` | Deleted | ? deleted ? |
| `http://bing.com/` | ? deleted ? | Deleted |

Reply:
Recommend: Remove the URI based APIs in favor of domain/path APIs which more clearly indicate what's actually being used. Except the Get which requires investigation if the scheme of the URI is respected.